tracer/src/Datadog.Trace/AppSec/Waf/NativeBindings/DdwafObjectStruct.cs

```diff
@@ -42,10 +42,51 @@ internal struct DdwafObjectStruct
         [FieldOffset(32)]
         public DDWAF_OBJ_TYPE Type;
 
-        internal IReadOnlyDictionary<string, string[]> Decode()
+        internal object Decode()
+        {
+            object res = Type switch
+            {
+                DDWAF_OBJ_TYPE.DDWAF_OBJ_STRING => Marshal.PtrToStringAnsi(Array, (int)NbEntries),
+                DDWAF_OBJ_TYPE.DDWAF_OBJ_SIGNED => IntValue,
+                DDWAF_OBJ_TYPE.DDWAF_OBJ_UNSIGNED => UintValue,
+                DDWAF_OBJ_TYPE.DDWAF_OBJ_ARRAY => DecodeArray<object>(),
+                DDWAF_OBJ_TYPE.DDWAF_OBJ_MAP => DecodeMap(),
+                _ => null
+            };
+
+            return res;
+        }
+
+        public List<T> DecodeArray<T>()
```

Comment:
I'm pretty sure this will only work if `T` is string or object, so may be safer to replace with `DecodeStringArray` and `DecodeObjectArray` methods. Otherwise maybe leave a comment saying this?

Reply:
In the end it is only used for object[], but the only difference is a cast when decoding each of the elements. I prefer to leave it generic, just in case in the future we need it for strings or other kind of elements.

Reply:
Yes the cast is the only difference, but you're setting up future uses of the method for failure since the cast will work for a few types. If you want to keep it, better to make it private and expose an method without the generic parameter.

Reply:
Made specific funcs in 61c8967e4722a95c3e525819e6adb8f0c4bdb4b8